metadata/src/artist.rs

```diff
@@ -111,10 +111,12 @@ pub struct Biographies(pub Vec<Biography>);
 impl_deref_wrapped!(Biographies, Vec<Biography>);
 
 #[derive(Debug, Clone)]
-pub struct ActivityPeriod {
-    pub start_year: i32,
-    pub end_year: i32,
-    pub decade: i32,
+pub enum ActivityPeriod {
+    Timespan {
+        start_year: i32,
+        end_year: Option<i32>,
+    },
+    Decade(i32),
```

Comment:
Would it make sense to make these three numbers unsigned (e.g. `u16` or even less) to potentially save space?

Reply:
I'm pretty sure we could. Probably Spotify sends everything as `i32` because there are no 16-bit integers in protobuf?

Reply:
Apparently, there would at least be [unsigned integers](https://developers.google.com/protocol-buffers/docs/proto#scalar). I'll leave the decision up to you: Do we want to not fail if Spotify sends us strange (negative) values? If we decide to go for conversion to `u16`, should this be fallible or is it OK to use `as u16` there (which would of course panic for negative values).

Reply:
Let’s return an error, certainly not panic in a library.